pkg/lang/capabilities.go

```diff
@@ -52,64 +57,87 @@ func CompositePreprocessor(preprocessors ...CommentPreprocessor) CommentPreproce
 //
 // To do this, you provide a `sitterQuery` that looks for comments nodes that contain the `klotho::` annotations,
 // as well as a preprocessor that runs over each of those nodes.
 //
 // If a source file contains multiple comment nodes in a row (as identified by having equal `.Type()`s), those comments
 // will be preprocessed individually, but then merged into a single annotation.
-func NewCapabilityFinder(sitterQuery string, preprocessor CommentPreprocessor) core.CapabilityFinder {
+func NewCapabilityFinder(sitterQuery string, preprocessor CommentPreprocessor, mergePredicate MergeCommentsPredicate) core.CapabilityFinder {
 	return &capabilityFinder{
-		sitterQuery:  sitterQuery,
-		preprocessor: preprocessor,
+		sitterQuery:            sitterQuery,
+		preprocessor:           preprocessor,
+		mergeCommentsPredicate: mergePredicate,
 	}
 }
 
-// FindAllCapabilities finds all of the annotations (ie, capabilities) in a SourceFile.
+// FindAllCapabilities finds all the annotations (ie, capabilities) in a SourceFile.
 func (c *capabilityFinder) FindAllCapabilities(f *core.SourceFile) (core.AnnotationMap, error) {
 	var merr multierr.Error
 	capabilities := make(core.AnnotationMap)
-	for _, block := range c.findAllCommentsBlocks(f) {
-		cap, err := annotation.ParseCapability(block.comment)
-		if cap == nil {
-			continue
-		}
-		annotation := &core.Annotation{Capability: cap, Node: block.node}
-		if err != nil {
-			merr.Append(core.NewCompilerError(f, annotation, errors.Wrap(err, "error parsing annotation")))
-			continue
+	for _, block := range c.findAllCommentBlocks(f) {
+		caps, err := annotation.ParseCapabilities(block.comment)
+		for i, cap := range caps {
+			var node *sitter.Node
+			if i == 0 {
+				node = block.annotatedNode
+			}
+			annot := &core.Annotation{Capability: cap, Node: node}
+			if err != nil {
+				merr.Append(core.NewCompilerError(f, annot, errors.Wrap(err, "error parsing annotation")))
+				continue
+			}
+			capabilities.Add(annot)
 		}
-		capabilities.Add(annotation)
 	}
 	return capabilities, merr.ErrOrNil()
 }
 
-func (c *capabilityFinder) findAllCommentsBlocks(f *core.SourceFile) []*commentBlock {
+func (c *capabilityFinder) findAllCommentBlocks(f *core.SourceFile) []*commentBlock {
 	const fullCaptureName = "fullQueryCaptureForFindAllCommentsBlocks" // please don't use this in your query ;)
 	queryString := fmt.Sprintf(`(%s) @%s`, c.sitterQuery, fullCaptureName)
 	nextMatch := query.Exec(f.Language, f.Tree().RootNode(), queryString)
 
-	blocks := []*commentBlock{}
-	combineWithPrevious := false
+	var blocks []*commentBlock
 	for {
 		match, found := nextMatch()
 		if !found || match == nil {
 			break
 		}
 		capture := match[fullCaptureName]
-		comment := capture.Content()
-		comment = c.preprocessor(comment)
+		comment := c.preprocessor(capture.Content())
+		annotatedNode := capture.NextNamedSibling()
+		if annotatedNode != nil && strings.Contains(annotatedNode.Type(), "comment") {
```

Comment:
Checking the node type is language specific. Maybe check the `c.sitterQuery` against it?

Reply:
I guess I can execute the full query and check if the next match == the annotated node